src/main/kotlin/senior/joinu/candid/IDLParser.kt

```diff
@@ -287,19 +281,22 @@ object IDLGrammar : Grammar<IDLProgram>() {
     private val pNatVal: Parser<IDLToken.NatVal> by pNatDec or pNatHex
 }
 
-interface IDLTextToken {
+interface IDLName {
     val value: String
 }
 
 sealed class IDLToken {
     sealed class NatVal : IDLToken() {
-        class Dec(val value: Int) : NatVal()
-        class Hex(val value: String) : NatVal()
+        class Dec(val value: Int) : NatVal() {
+            companion object { const val pattern: String = "[\\-+]?[0-9][_0-9]*" }
+        }
+        class Hex(val value: String) : NatVal() {
+            companion object { const val pattern: String = "0x[0-9a-fA-F][_0-9a-fA-F]*" }
+        }
     }
 
-    class TextVal(override val value: String) : IDLToken(),
-        IDLTextToken
-
-    class Name(override val value: String) : IDLToken(),
-        IDLTextToken
+    data class TextVal(override val value: String) : IDLName {
```

Comment:
Please make it also inherit the `IDLToken()` parent.

Reply:
It is needed to make `when` expression work correctly.

Reply:
Done.